### PR TITLE
perf(ripple): faster cold hydration and server rendering (news benchmark)

### PR DESCRIPTION
**Pull request by leonidaz**
## Summary

Continues the per-benchmark optimization series (js-framework, chat-stream, effectful-list, memo-wall, portal-swarm) with `benchmarks/news`: SSR render vs Svelte and cold hydration vs Inferno.

Both ops are cold-start measurements. Chrome tracing showed ~55% of Ripple's hydrate window was V8 lazily parsing and compiling the runtime functions first called during `hydrate()`; execution time was already at parity with Inferno, and the cost tracks the *number* of functions compiled more than bytes (72 vs Inferno's 44 at similar size). Cold SSR was dominated by `flat(Infinity).join('')` and per-block `Output` allocations while the code still runs in the interpreter tier.

### Client runtime
- `hydrate()` accepts `rootBoundary: false` like `mount()`, stepping over the server's root markers itself instead of rendering under a `try` block (the single largest function compiled on a cold page). The news fixture uses it, matching the other targets which have no root boundary.
- Hydration cursor steps (`hydrate_first_child`, `hydrate_next_sibling`, `hydrate_append`) live in `hydration.js` and assign the cursor directly through the exported `firstChild`/`nextSibling` getters.
- A list's first render never compiles the diff (`reconcile_by_ref_diff` split, like the keyed one); a template's clone path leaves the per-template closure (`clone_template`); `run_block` keeps re-run/teardown handling in `prepare_rerun`/`register_teardown`; `pop_component` defers to `create_deferred_effects`; `finish_dependencies` is guarded; `push_block`, `assign_nodes`, `create_component_ctx`, `run_component`, `rehydrate_anchor`, `normalize_props` are inlined at their hot call sites; `render_value` loops instead of recursing through `render_tsrx_element`.
- Removing the server's inline styles starts from an animation frame rather than inside `mount()`/`hydrate()` (the removal itself already waited for a frame after the stylesheets loaded).

### Compiler
- An element's lone text child hydrates in place through a new `hydrate_text` op: the cursor stays on the element, so no `pop()` is emitted and `append()` skips its `contains()` descent check.
- A `@for` whose `index` nothing reads no longer sets `IS_INDEXED`, so no tracked index is allocated per item.

### Server
- `render()` joins its buffer tree with a plain recursive walk instead of `flat(Infinity).join('')` (30–45 µs steady, spiking to hundreds of µs during tier-up, down to 4–9 µs).
- Ripple-owned `escape()`: strings without `&`, `<` (or `"` in attributes) return as-is after one regex test; anything else goes through `@tsrx/core`'s escaper.
- `Output` allocates its stream, css and async-operation collections only for the root or on first use; `push_component` uses a shared noop; `render()` no longer spreads its options; the nesting-warning set is created lazily.

### Fixture
- `App.tsrx` types the articles so every `{a.*}` hole lowers to `set_text` instead of an `expression()` block with hydration markers (verified against the compiled output).

## Results

Official harness (`pnpm bench news`), same machine:

| Op | Before | After | Best competitor |
| --- | --- | --- | --- |
| hydrate | 2.99 ms | 2.26 ms | Inferno 2.40 ms |
| ssr_render | 0.13 ms | 0.05 ms | Svelte 0.07 ms |

Interleaved fresh-context A/B (mean of 30): Ripple 2.14 ms vs Inferno 2.24 ms. `ssr-throughput` news-50: 0.146 → 0.065 ms (octane-tsrx 0.081). Functions compiled during a cold hydrate: 83 → 61.

## Verification
- `pnpm test` (2241 tests), typecheck of `ripple` and `@tsrx/ripple` (one pre-existing `hmr.js` error untouched), `pnpm format:check`, `pnpm changeset:check`.
- `pnpm bench --quick` gates pass for js-framework, chat-stream, memo-wall, effectful-list, portal-swarm, todomvc and ssr-throughput.
- Hydration compiled test fixtures regenerated by the test setup (`hydrate_text`, no per-text `pop`, list flags).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Touches core client hydration/mount and SSR output assembly with new root-boundary semantics; mitigated by broad hydration tests and regenerated compiled fixtures, but regressions would surface as mismatches or streaming edge-case bugs.
> 
> **Overview**
> This PR speeds up **cold hydration** and **server rendering** by shrinking what V8 must compile on first paint and by trimming hot-path work on the server.
> 
> **Hydration and mount:** `hydrate()` now accepts **`rootBoundary: false`** (matching `mount()`), stepping over a plain `<!--[-->` marker instead of going through the default root `try` block—except when streaming left a pending root slot, which still requires a boundary. SSR inline style removal is **`requestAnimationFrame`**-scheduled so mount/hydrate no longer touch stylesheets synchronously.
> 
> **Client runtime:** Hydration cursor logic is centralized in **`hydration.js`** (direct `firstChild`/`nextSibling` getters, `set_hydration`, `hydrate_append`, `hydrate_text`). Cold paths avoid extra compiled code: list **first render** skips the reconcile diff function, **template cloning** is split from the hydrate closure, **`run_block`** teardown/rerun helpers are separated, and several hot sites are inlined. The compiler emits **`hydrate_text`** for an element’s only text child (no **`pop()`**), and **`IS_INDEXED`** is omitted when nothing reads `@for`’s `index`.
> 
> **Server:** Buffer assembly uses a **recursive `flatten_buffer` walk** instead of `flat(Infinity).join('')`, **`escape()`** fast-paths safe strings, and **`Output`** defers allocating stream/CSS/async collections until the root or first use.
> 
> **Benchmarks/tests:** The news **`App`** types articles so string holes compile to **`set_text`**; fixtures and tests cover **`rootBoundary: false`** and streamed root hydration.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b09e0edda760564cfc284e3f414fd2317eb76144. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/ripple/src/runtime/index-client.js [hunk 1]
```diff
@@ -159,19 +157,28 @@ export function hydrate(component, options) {
...
 
+		const root_boundary = options.rootBoundary;
+
 		_root = root(() => {
+			if (root_boundary === false) {
+				// The root boundary's own hydration walk: consume the `<!--[-->`
+				// marker and render against it, as `try_block` does for the root.
+				hydrate_next();
+				render_component(component, /** @type {Comment} */ (anchor), props);
+				return;
```
[the remark this answers is not in the document]
Comment (leonidaz) [reply packages/ripple/src/runtime/index-client.js:179]: Good catch. Kept the `rootBoundary: false` opt-out but it now applies only when the root marker is a plain `<!--[-->`; a streamed `<!--[?N-->` / `<!--[!N-->` shell hydrates under the default boundary regardless, since only a boundary can adopt the fallback and activate the chunk. Fixed in b19d4d8d6 with tests for both shapes.
